common/db_store/admin_db.py

```diff
@@ -0,0 +1,138 @@
+import pymongo
+from abc import ABC, abstractmethod
+from common.db_store.error import DBError
+
+
+class DBAdmin(ABC):
+
+    @abstractmethod
+    def create_index(self, collection, index_key, **kwargs):
+        pass
+
+    @abstractmethod
+    def list_indexes(self, collection, **kwargs):
+        pass
+
+    @abstractmethod
+    def drop_index(self, collection, index_name, **kwargs):
+        pass
+
+    @abstractmethod
+    def close_connection(self):
+        pass
+
+    @abstractmethod
+    def create_timeseries_collection(self, collection, timeField, **kwargs):
+        pass
+
+    @abstractmethod
+    def get_list_of_collections(self):
+        pass
+
+
+class MongoDBAdmin(DBAdmin):
+
+    def __init__(self, endpoint: str, db_name: str, **kwargs) -> None:
+        """
+        Initialize DB connection.
+        Args -
+            endpoint (str): MongoDB server endpoints.
+                Ex. mongodb://mongo-0.mongo,mongo-1.mongo,mongo-2.mongo:27017
+            db_name (str): Name of database.
+        """
+        try:
+            self._client = pymongo.MongoClient(endpoint)
+            self._db = self._client[db_name]
+        except Exception as e:
+            # Log.error(f"Unable to connect database server endpoints \
```

Comment:
remove if not required

Reply:
Currently in cortx-halo logging is not enabled, & We're planning to enable it in near future, So it will be good to have these comments.

Reply:
could you add a TODO statement with it, regarding the same